internal/storage/v1/elasticsearch/spanstore/reader.go

```diff
@@ -328,20 +328,36 @@ func bucketToStringArray(buckets []*elastic.AggregationBucketKeyItem) ([]string,
 	return strings, nil
 }
 
+func bucketToTraceIDArray(buckets []*elastic.AggregationBucketKeyItem) ([]dbmodel.TraceID, error) {
+	ids := make([]dbmodel.TraceID, len(buckets))
+	for i, keyitem := range buckets {
+		str, ok := keyitem.Key.(string)
+		if !ok {
+			return nil, errors.New("non-string key found in aggregation")
+		}
+		ids[i] = dbmodel.TraceID(str)
+	}
+	return ids, nil
+}
+
 // FindTraces retrieves traces that match the traceQuery
 func (s *SpanReader) FindTraces(ctx context.Context, traceQuery *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
 	ctx, span := s.tracer.Start(ctx, "FindTraces")
 	defer span.End()
 
-	uniqueTraceIDs, err := s.FindTraceIDs(ctx, traceQuery)
+	uniqueTraceIDs, err := s.FindTraceIDs(ctx, esTraceQueryParamsFromSpanStoreTraceQueryParams(traceQuery))
```

Comment:
the query translation should be only in reader-v1, not here. You can change the signature of the input without changing the output, so that you don't have to fully implement FindTraces in this PR